superbench/analyzer/result_summary.py

```diff
@@ -0,0 +1,244 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+"""A module for Result Summary."""
+
+import re
+from pathlib import Path
+
+import pandas as pd
+
+from superbench.common.utils import logger
+from superbench.analyzer import file_handler
+from superbench.analyzer.summary_op import SummaryOp, SummaryType
+from superbench.analyzer import RuleBase
+from superbench.analyzer import data_analysis
+
+
+class ResultSummary(RuleBase):
+    """Result summary class."""
+    def _check_rules(self, rule, name):
+        """Check whether the formart of the rule is valid.
+
+        Args:
+            rule (dict): the rule
+            name (str): the rule name
+
+        Returns:
+            dict: the rule for the metric
+        """
+        # check if rule is supported
+        super()._check_and_format_rules(rule, name)
+        if 'metrics' not in rule:
+            logger.log_and_raise(exception=Exception, msg='{} lack of metrics'.format(name))
+        if 'statistics' not in rule:
+            logger.log_and_raise(exception=Exception, msg='{} lack of function'.format(name))
+        # convert single statistic str to list
+        if not isinstance(rule['statistics'], list):
+            rule['statistics'] = [rule['statistics']]
+        # check statistics format, should be SummaryType or p\d\d?
+        for function in rule['statistics']:
+            if not (re.fullmatch(r'p\d\d?', function) or isinstance(SummaryType(function), SummaryType)):
+                logger.log_and_raise(exception=Exception, msg='{} invalid function name'.format(name))
```

Comment:
Here should be `format(function)` not `format(name)`?
But if `SummaryType(function)` raise Exception, this line will not get the chance to execute.

Reply:
1. the name is a tip for rule name only, but I can add function name in the end. 
2. yes, add a try for that, and